server/apps/immich/src/modules/immich-jwt/strategies/public-share.strategy.ts

```diff
@@ -0,0 +1,54 @@
+import { UserEntity } from '@app/database';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ShareService } from 'apps/immich/src/api-v1/share/share.service';
+import { IStrategyOptions, Strategy } from 'passport-http-header-strategy';
+import { Repository } from 'typeorm';
+
+export const PUBLIC_SHARE_STRATEGY = 'public-share';
+
+const options: IStrategyOptions = {
+  header: 'x-immich-share-key',
+  param: 'key',
+};
+
+export class PublicUser extends UserEntity {
+  isPublicUser?: boolean;
+}
+
+@Injectable()
+export class PublicShareStrategy extends PassportStrategy(Strategy, PUBLIC_SHARE_STRATEGY) {
+  constructor(
+    private shareService: ShareService,
+    @InjectRepository(UserEntity)
+    private usersRepository: Repository<UserEntity>,
+  ) {
+    super(options);
+  }
+
+  async validate(key: string): Promise<PublicUser> {
+    const validatedLink = await this.shareService.getSharedLinkByKey(key);
+
+    if (validatedLink.expiresAt) {
+      const now = new Date().getTime();
+      const expiresAt = new Date(validatedLink.expiresAt).getTime();
+
+      if (now > expiresAt) {
+        throw new BadRequestException('Expired link');
```

Comment:
I think all the errors in here should be UnauthorizedExceptions since that's the error. This is not a login request, but a request to access a resource, so like `GET /asset/123`. It wasn't a bad request (the request is not malformed), it's an unauthorized request (the authentication is invalid).